lib/rets/client.rb

```diff
@@ -278,11 +278,28 @@ def metadata
     def retrieve_metadata
       raw_metadata = {}
       Metadata::METADATA_TYPES.each {|type|
-        raw_metadata[type] = retrieve_metadata_type(type)
+        raw_metadata[type] = filter_metadata(type, retrieve_metadata_type(type))
       }
       raw_metadata
     end
 
+    def filter_metadata(metadata_type, xml)
+      return xml unless @options[:metadata_filter]
+
+      types_to_filter = @options[:metadata_filter][:types_to_filter]
```

Comment:
What type would you not want to filter? wondering if this really needs to be a required option for all mlses or if this can have a sensible default that almost everyone uses.